File: Manual Copies/108-Convert-Sorted_Array_To_Binary_Search_Tree/Solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
 var sortedArrayToBST = function(nums) {
  // Base case if array.length === 0
  if (nums.length === 0) {
    return null;
  }
  // Create a new node
  let node = new TreeNode;
  // Set the value of the node to the middle of the array
  node.val = nums[Math.floor(nums.length/2)];
  // Set the left and right of the node to the left and right halves of the array recursively
  node.left = sortedArrayToBST(nums.slice(0, Math.floor(nums.length/2)));
  node.right = sortedArrayToBST(nums.slice(Math.floor(nums.length/2) + 1, nums.length));
  // Return node
  return node;
};

File: Manual Copies/108-Convert-Sorted_Array_To_Binary_Search_Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
  public TreeNode sortedArrayToBST(int[] nums) {
      // Base case, if nums is empty, return null
      if (nums.length == 0) {
          return null;
      }
      // Create a new TreeNode
      TreeNode node = new TreeNode();
      // Set the value of the TreeNode to the middle element of nums
      node.val = nums[(int) Math.floor(nums.length / 2)];
      // Recursively call sortedArrayToBST on the left and right halves of nums
      node.left = sortedArrayToBST(Arrays.copyOfRange(nums, 0, (int) Math.floor(nums.length / 2)));
      node.right = sortedArrayToBST(Arrays.copyOfRange(nums, (int) Math.floor(nums.length / 2) + 1, nums.length));
      // Return the TreeNode
      return node;
  }
}